### PR TITLE
Allow sending time_to_live field

### PR DESCRIPTION
**Pull request by kanes115**
This PR relies on merging [this PR](https://github.com/rslota/pigeon/pull/1) to Pigeon.

This PR lets mongooseim send `time_to_live` option in notifications.

### CODE REVIEW

File: test/mongoose_push_test.exs [hunk 1]
```diff
@@ -147,12 +147,14 @@ defmodule MongoosePushTest do
...
         sound: string(min: 3, max: 15, chars: :ascii),
         click_action: string(min: 3, max: 15, chars: :ascii),
+        time_to_live: int(min: 0, max: 4500),
         priority: choose(from: [value(:normal), value(:high)]),
       ], repeat_for: 10 do
 
       notification =
         %{:service => :fcm,
           :priority => priority,
+          :time_to_live => time_to_live,
```
Comment (rslota) [inline test/mongoose_push_test.exs:157]: There is no actual check whether this is correctly set in the notification. We need that tested.

File: lib/mongoose_push.ex [hunk 2]
```diff
@@ -63,6 +63,7 @@ defmodule MongoosePush do
         request
         |> Map.put(:alert, request[:alert])
         |> Map.put(:data, request[:data])
+        |> Map.put(:time_to_live, request[:time_to_live])
```
Comment (rslota) [inline lib/mongoose_push.ex]: Why do we need it here? `alert` and `data` need this to ensure those keys are in `request` map, but I don't see any logic in this PR that would require `time_to_live` to be present regardless of whether it's set or not.

**Review by rslota — CHANGES_REQUESTED** [review]
The code looks good in general. Tests and documentation (in `README.md`) seem to be missing though.

File: test/mongoose_push_test.exs [hunk 3]
```diff
@@ -182,6 +184,7 @@ defmodule MongoosePushTest do
       assert notification.alert[:tag] == fcm_data["tag"]
       assert notification.alert[:sound] == fcm_data["sound"]
       assert notification[:data] == fcm_custom
+      assert notification[:time_to_live] == fcm_data["time_to_live"]
```
Comment (rslota) [inline test/mongoose_push_test.exs]: According to FCM documentation, `time_to_live` should be at the top level, among things like `priority`. This means that this check should be: 

```
assert notification[:time_to_live] == fcm_request["request_data"]["time_to_live"]
```

**Review by rslota — CHANGES_REQUESTED** [review]
The code looks good to me, but the documentation in `README.md` is still missing. We should add this "field" to list of supported options with a note regarding APNS incompatibility.

**Comment by kanes115** [comment]
@rslota Hope this kind of note is sufficient (the best source is the FCM documentation in this case, isn't it?)

**Review by rslota — APPROVED** [review]

**Pull request merged**